connectors/sources/microsoft_teams.py

```diff
@@ -0,0 +1,1316 @@
+
+        Returns:
+            dictionary: Content document with _id, _timestamp and attachment content
+        """
+        document_size = int(user_attachment["size_in_bytes"])
+
+        if not (doit and document_size):
+            return
+        filename = user_attachment["name"]
+        if not self.verify_filename_for_extraction(filename=filename):
+            return
+        if user_attachment["size_in_bytes"] > MAX_FILE_SIZE:
+            self._logger.warning(
+                f"File size {document_size} of file {filename} is larger than {MAX_FILE_SIZE} bytes. Discarding file content"
+            )
+            return
+        document = {
+            "_id": user_attachment["id"],
+            "_timestamp": user_attachment["_timestamp"],
+        }
+        self._logger.debug(f"Downloading {filename}")
+        attachment_content = await self._download_content_for_attachment(
+            partial(self.client.download_item, download_url),
+            original_filename=filename,
+        )
+        document["_attachment"] = attachment_content
+
+        return document
+
+    def format_attachment_names(self, attachments):
+        return ",".join(
+            attachment.get("name")
+            for attachment in attachments
+            if attachment.get("name", "")
+        )
+
+    async def get_messages(
+        self, message, document_type=None, chat=None, channel_name=None, members=None
+    ):
+        if not message.get("deletedDateTime") and (
+            "unknownFutureValue" not in message.get("messageType")
+        ):
+            if message_content := html_to_text(
+                html=message.get("body", {}).get("content")
+            ):
+                if document_type == UserEndpointName.CHATS_MESSAGE.value:
+                    message_document = await self.format_user_chat_messages(
+                        chat=chat,
+                        message=message,
+                        message_content=message_content,
+                        members=members,
+                    )
+                    await self.queue.put(
+                        (
+                            self.format_doc(
+                                item=message_document,
+                                document_type=self.schema.chat_messages,
+                                document={"type": document_type},
+                            ),
+                            None,
+                        )
+                    )
+                else:
+                    await self.queue.put(
+                        (
+                            self.format_channel_message(
+                                item=message,
+                                channel_name=channel_name,
+                                message_content=message_content,
+                            ),
+                            None,
+                        )
+                    )
+
+    async def user_chat_meeting_recording(self, message):
```

Comment:
Is this func updating a doc? If so, I think a verb should be added to the name, e.g. `update_user_chat_meeting_recording`. With its current naming, I assumed it was representing an object.

The same should be applied to the other funcs in this class that are doing something.

Reply:
This function is creating a new doc everytime.